lib/period.spec.js

```diff
@@ -243,6 +243,37 @@ describe('periods', () => {
     done();
   });
 
+  it('should allow to get proof from period with prevPeriod.', (done) => {
+    const prevTx = '0x7777777777777777777777777777777777777777777777777777777777777777';
+    const value = 99000000;
+    const color = 1337;
+    const transfer = Tx.transfer(
+      [new Input(new Outpoint(prevTx, 0))],
+      [new Output(value / 2, ADDR, color), new Output(value / 2, ADDR, color)],
+    );
+    transfer.sign([PRIV]);
+
+    const blocks = [];
+    let block;
+    for (let i = 0; i < 30; i++) {
+      block = new Block(i).addTx(Tx.deposit(i, value, ADDR, color));
+      blocks.push(block);
+    }
+    block = new Block(31).addTx(transfer);
+    blocks.push(block);
+
+    const period = new Period("0x8b04de057fe524a3118eb7c8e14a2e55323c67fd7b6080583d1047b700b2d674", blocks);
+    period.setValidatorData(slotId, ADDR);
+    const periodAfter = new Period(period.periodRoot(), blocks);
+    periodAfter.setValidatorData(slotId, ADDR);
+    periodAfter.usePrevPeriod();
+    
+    const proof = periodAfter.prevPeriodProof();
+    expect(Period.verifyPrevPeriodProof(proof)).to.eql(periodAfter.periodRoot());
```

Comment:
this isn't really checks that `period.periodRoot()` was used in `proof`, no?

Reply:
See below.